regex-syntax/src/ast/mod.rs

```diff
@@ -153,6 +153,9 @@ pub enum ErrorKind {
     /// The range provided in a counted repetition operator is invalid. The
     /// range is invalid if the start is greater than the end.
     RepetitionCountInvalid,
+    /// An opening `{` was not followed by a valid decimal value.
+    /// For example, `x{}` or `x{]}` would fail.
+    RepetitionQuantifierDecimalMissing,
```

Comment:
With this change, it looks like `DecimalEmpty` is no longer used. This is a bit weird, but I see why it makes sense to do that. Could you update the docs of `DecimalEmpty` to say something like:

```rust
/// Note that this error variant is no longer used. Namely, a decimal
/// number can only appear as a repetition quantifier. When the number
/// in a repetition quantifier is empty, then it gets its own specialized
/// error, `RepetitionQuantifierDecimalMissing`.

Reply:
Also, could you name this `RepetitionCountDecimalEmpty` in order to be consistent with other error variants?